---
title: "CS1017 chyby kompilátoru"
ms.date: 07/20/2015
ms.prod: .net
ms.technology:
- devlang-csharp
ms.topic: article
f1_keywords:
- CS1017
helpviewer_keywords:
- CS1017
ms.assetid: e0902e8a-b042-4711-a8a6-83456a3f88cb
caps.latest.revision: 
author: BillWagner
ms.author: wiwagn
ms.openlocfilehash: adfebf5f696090aa7355025e4c2cba2240e1a6c3
ms.sourcegitcommit: 83dd5ec003e788ccb3e33f3412a7af39ae347646
ms.translationtype: MT
ms.contentlocale: cs-CZ
ms.lasthandoff: 03/15/2018
---
# <a name="compiler-error-cs1017"></a><span data-ttu-id="562d3-102">CS1017 chyby kompilátoru</span><span class="sxs-lookup"><span data-stu-id="562d3-102">Compiler Error CS1017</span></span>
<span data-ttu-id="562d3-103">Klauzule catch nelze provést v klauzuli obecné catch příkazu try</span><span class="sxs-lookup"><span data-stu-id="562d3-103">Catch clauses cannot follow the general catch clause of a try statement</span></span>  
  
 <span data-ttu-id="562d3-104">A `catch` blok, který nevyžaduje žádné parametry musí být poslední v řadě `catch` bloky.</span><span class="sxs-lookup"><span data-stu-id="562d3-104">A `catch` block that does not take any parameters must be the last in a series of `catch` blocks.</span></span> <span data-ttu-id="562d3-105">Další informace o výjimky najdete v tématu [příkazy zpracování výjimek](../../csharp/language-reference/keywords/exception-handling-statements.md).</span><span class="sxs-lookup"><span data-stu-id="562d3-105">For more information on exceptions, see [Exception Handling Statements](../../csharp/language-reference/keywords/exception-handling-statements.md).</span></span>  
  
## <a name="example"></a><span data-ttu-id="562d3-106">Příklad</span><span class="sxs-lookup"><span data-stu-id="562d3-106">Example</span></span>  
 <span data-ttu-id="562d3-107">Následující ukázka generuje CS1017:</span><span class="sxs-lookup"><span data-stu-id="562d3-107">The following sample generates CS1017:</span></span>  
  
```csharp  
// CS1017.cs  
using System;  
  
namespace x  
{  
    public class b : Exception  
    {  
    }  
  
    public class a  
    {  
        public static void Main()  
        {  
            try  
            {  
            }  
  
            catch   // CS1017, must be last catch  
            {  
            }  
  
            catch(b)  
            {  
                throw;  
            }  
        }  
    }  
}  
```
